internal/provider/resource_ssh_key_test.go

```diff
@@ -0,0 +1,157 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/oxidecomputer/oxide.go/oxide"
+)
+
+type resourceSSHKeyConfig struct {
+	BlockName   string
+	Name        string
+	Description string
+	PublicKey   string
+}
+
+var resourceSSHKeyConfigTpl = `
+resource "oxide_ssh_key" "{{.BlockName}}" {
+  name        = "{{.Name}}"
+  description = "{{.Description}}"
+  public_key  = "{{.PublicKey}}"
+  timeouts = {
+    read   = "1m"
+    create = "3m"
+    delete = "2m"
+    update = "4m"
+  }
+}
+`
+
+var resourceSSHKeyUpdateConfigTpl = `
+resource "oxide_ssh_key" "{{.BlockName}}" {
+  name        = "{{.Name}}"
+  description = "{{.Description}}"
+  public_key  = "{{.PublicKey}}"
+  timeouts = {
+    read   = "1m"
+    create = "3m"
+    delete = "2m"
+    update = "4m"
+  }
+}
+`
+
+func TestAccResourceSSHKey_full(t *testing.T) {
+	sshKeyName := newResourceName()
+	description := "An SSH key."
+	publicKey := "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIE1clIQrzlQNqxgvpCCUFFOcTTFDOaqV+aocfsDZvxqB"
+	blockName := newBlockName("ssh_key")
+	resourceName := fmt.Sprintf("oxide_ssh_key.%s", blockName)
+	config, err := parsedAccConfig(
+		resourceSSHKeyConfig{
+			BlockName:   blockName,
+			Name:        sshKeyName,
+			Description: description,
+			PublicKey:   publicKey,
+		},
+		resourceSSHKeyConfigTpl,
+	)
+	if err != nil {
+		t.Errorf("error parsing config template data: %e", err)
+	}
+
+	sshKeyNameUpdated := sshKeyName + "-updated"
+	configUpdate, err := parsedAccConfig(
+		resourceSSHKeyConfig{
+			BlockName:   blockName,
+			Name:        sshKeyNameUpdated,
+			Description: description,
+			PublicKey:   publicKey,
+		},
+		resourceSSHKeyUpdateConfigTpl,
+	)
+	if err != nil {
+		t.Errorf("error parsing config template data: %e", err)
+	}
```

Comment:
Updates aren't supported! :smile:

Reply:
You know, I wasn't sure about that logic. Part of me thought to leave it to cover the destroy and create case, but I didn't dig into it too much. Thank you!